From 20aab6636777e0aa92a149d82db5cc5b51a4ba2a Mon Sep 17 00:00:00 2001
From: Patrick Rohr <prohr@google.com>
Date: Thu, 22 Dec 2022 14:14:27 -0800
Subject: [PATCH] cronet: fix cronet build with enable_base_tracing = false

Cherry-pick crrev/c/3989184 which has not been accepted into upstream.

Test: none
Change-Id: Ibda168d1bdb1966ac64cea06525d013f7838632c
---
 net/base/network_delegate.cc                                   | 2 +-
 net/cert/crl_set.cc                                            | 2 +-
 net/cert/multi_threaded_cert_verifier.cc                       | 2 +-
 net/cert/x509_certificate.cc                                   | 2 +-
 net/disk_cache/blockfile/backend_impl.cc                       | 2 +-
 net/disk_cache/blockfile/eviction.cc                           | 2 +-
 net/dns/host_cache.cc                                          | 2 +-
 net/dns/host_resolver_manager.cc                               | 2 +-
 net/dns/host_resolver_system_task.cc                           | 2 +-
 net/http/http_cache_transaction.cc                             | 2 +-
 net/http/http_response_headers.cc                              | 2 +-
 net/http/http_response_headers.h                               | 2 +-
 net/http/http_stream_factory_job.cc                            | 2 +-
 net/log/trace_net_log_observer.cc                              | 2 +-
 net/log/trace_net_log_observer.h                               | 2 +-
 net/nqe/network_quality_estimator.cc                           | 2 +-
 net/quic/quic_stream_factory.cc                                | 2 +-
 net/socket/client_socket_handle.cc                             | 2 +-
 net/socket/connect_job.cc                                      | 2 +-
 net/socket/socket_posix.cc                                     | 2 +-
 net/socket/socks5_client_socket.cc                             | 2 +-
 net/socket/ssl_client_socket_impl.cc                           | 2 +-
 net/socket/ssl_connect_job.cc                                  | 2 +-
 net/socket/tcp_socket_posix.cc                                 | 3 +--
 net/socket/transport_connect_job.cc                            | 2 +-
 net/socket/udp_socket_posix.cc                                 | 3 +--
 net/spdy/spdy_session.cc                                       | 2 +-
 net/spdy/spdy_session_pool.cc                                  | 2 +-
 .../android/embedded_test_server_android.cc                    | 2 +-
 net/test/embedded_test_server/controllable_http_response.cc    | 2 +-
 net/url_request/url_request_http_job.cc                        | 2 +-
 url/gurl.cc                                                    | 2 +-
 url/gurl.h                                                     | 2 +-
 url/origin.cc                                                  | 2 +-
 url/origin.h                                                   | 2 +-
 35 files changed, 35 insertions(+), 37 deletions(-)

diff --git a/net/base/network_delegate.cc b/net/base/network_delegate.cc
index e119ca3d789be..673f4ecc7b805 100644
--- a/net/base/network_delegate.cc
+++ b/net/base/network_delegate.cc
@@ -8,7 +8,7 @@
 
 #include "base/logging.h"
 #include "base/ranges/algorithm.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "net/base/load_flags.h"
 #include "net/base/net_errors.h"
 #include "net/base/trace_constants.h"
diff --git a/net/cert/crl_set.cc b/net/cert/crl_set.cc
index e7d5066cbcf79..8216f43ed7bd9 100644
--- a/net/cert/crl_set.cc
+++ b/net/cert/crl_set.cc
@@ -10,7 +10,7 @@
 #include "base/json/json_reader.h"
 #include "base/strings/string_piece.h"
 #include "base/time/time.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "base/values.h"
 #include "crypto/sha2.h"
 #include "net/base/trace_constants.h"
diff --git a/net/cert/multi_threaded_cert_verifier.cc b/net/cert/multi_threaded_cert_verifier.cc
index 1e61c4818fbed..2f32f513ad1f0 100644
--- a/net/cert/multi_threaded_cert_verifier.cc
+++ b/net/cert/multi_threaded_cert_verifier.cc
@@ -11,7 +11,7 @@
 #include "base/memory/weak_ptr.h"
 #include "base/task/thread_pool.h"
 #include "base/threading/thread_restrictions.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "crypto/crypto_buildflags.h"
 #include "net/base/net_errors.h"
 #include "net/base/trace_constants.h"
diff --git a/net/cert/x509_certificate.cc b/net/cert/x509_certificate.cc
index be2c1d3aaded9..4b37751ff71fc 100644
--- a/net/cert/x509_certificate.cc
+++ b/net/cert/x509_certificate.cc
@@ -20,7 +20,7 @@
 #include "base/strings/string_piece.h"
 #include "base/strings/string_util.h"
 #include "base/time/time.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "build/build_config.h"
 #include "crypto/openssl_util.h"
 #include "net/base/ip_address.h"
diff --git a/net/disk_cache/blockfile/backend_impl.cc b/net/disk_cache/blockfile/backend_impl.cc
index a12a6787a558d..3b8f040c2879e 100644
--- a/net/disk_cache/blockfile/backend_impl.cc
+++ b/net/disk_cache/blockfile/backend_impl.cc
@@ -31,7 +31,7 @@
 #include "base/threading/thread_task_runner_handle.h"
 #include "base/time/time.h"
 #include "base/timer/timer.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "net/base/net_errors.h"
 #include "net/disk_cache/backend_cleanup_tracker.h"
 #include "net/disk_cache/blockfile/disk_format.h"
diff --git a/net/disk_cache/blockfile/eviction.cc b/net/disk_cache/blockfile/eviction.cc
index a6db3780db26e..3b609892aa640 100644
--- a/net/disk_cache/blockfile/eviction.cc
+++ b/net/disk_cache/blockfile/eviction.cc
@@ -42,7 +42,7 @@
 #include "base/task/single_thread_task_runner.h"
 #include "base/threading/thread_task_runner_handle.h"
 #include "base/time/time.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "net/disk_cache/blockfile/backend_impl.h"
 #include "net/disk_cache/blockfile/disk_format.h"
 #include "net/disk_cache/blockfile/entry_impl.h"
diff --git a/net/dns/host_cache.cc b/net/dns/host_cache.cc
index a4d1399e3c411..0aa9eeae2bdfc 100644
--- a/net/dns/host_cache.cc
+++ b/net/dns/host_cache.cc
@@ -22,7 +22,7 @@
 #include "base/strings/string_number_conversions.h"
 #include "base/strings/string_piece.h"
 #include "base/time/default_tick_clock.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "base/types/optional_util.h"
 #include "base/value_iterators.h"
 #include "net/base/address_family.h"
diff --git a/net/dns/host_resolver_manager.cc b/net/dns/host_resolver_manager.cc
index 9db789cd87a24..3442ecfcf8a24 100644
--- a/net/dns/host_resolver_manager.cc
+++ b/net/dns/host_resolver_manager.cc
@@ -56,7 +56,7 @@
 #include "base/threading/thread_task_runner_handle.h"
 #include "base/time/default_tick_clock.h"
 #include "base/time/time.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "base/types/optional_util.h"
 #include "base/values.h"
 #include "build/build_config.h"
diff --git a/net/dns/host_resolver_system_task.cc b/net/dns/host_resolver_system_task.cc
index 3dfea30a98499..164a4d8cfbc23 100644
--- a/net/dns/host_resolver_system_task.cc
+++ b/net/dns/host_resolver_system_task.cc
@@ -13,7 +13,7 @@
 #include "base/task/task_traits.h"
 #include "base/task/thread_pool.h"
 #include "base/threading/scoped_blocking_call.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "base/types/pass_key.h"
 #include "dns_reloader.h"
 #include "net/base/net_errors.h"
diff --git a/net/http/http_cache_transaction.cc b/net/http/http_cache_transaction.cc
index 3ed7ea38ce9e0..4353ef8c17453 100644
--- a/net/http/http_cache_transaction.cc
+++ b/net/http/http_cache_transaction.cc
@@ -33,8 +33,8 @@
 #include "base/task/single_thread_task_runner.h"
 #include "base/threading/thread_task_runner_handle.h"
 #include "base/time/clock.h"
+#include "base/trace_event/base_tracing.h"
 #include "base/trace_event/common/trace_event_common.h"
-#include "base/trace_event/trace_event.h"
 #include "base/values.h"
 #include "crypto/secure_hash.h"
 #include "crypto/sha2.h"
diff --git a/net/http/http_response_headers.cc b/net/http/http_response_headers.cc
index 17d9b46cc3d2d..1546f48f62a85 100644
--- a/net/http/http_response_headers.cc
+++ b/net/http/http_response_headers.cc
@@ -27,7 +27,7 @@
 #include "base/strings/string_util.h"
 #include "base/strings/stringprintf.h"
 #include "base/time/time.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "base/values.h"
 #include "net/base/parse_number.h"
 #include "net/http/http_byte_range.h"
diff --git a/net/http/http_response_headers.h b/net/http/http_response_headers.h
index 5edf7025bc273..052e491e4377e 100644
--- a/net/http/http_response_headers.h
+++ b/net/http/http_response_headers.h
@@ -16,10 +16,10 @@
 #include "base/memory/ref_counted.h"
 #include "base/strings/string_piece.h"
 #include "base/time/time.h"
+#include "base/trace_event/base_tracing_forward.h"
 #include "net/base/net_export.h"
 #include "net/http/http_version.h"
 #include "net/log/net_log_capture_mode.h"
-#include "third_party/perfetto/include/perfetto/tracing/traced_value_forward.h"
 
 namespace base {
 class Pickle;
diff --git a/net/http/http_stream_factory_job.cc b/net/http/http_stream_factory_job.cc
index c4a4e046b8fc7..21114451c049f 100644
--- a/net/http/http_stream_factory_job.cc
+++ b/net/http/http_stream_factory_job.cc
@@ -23,7 +23,7 @@
 #include "base/strings/string_util.h"
 #include "base/task/single_thread_task_runner.h"
 #include "base/threading/thread_task_runner_handle.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "base/values.h"
 #include "build/build_config.h"
 #include "net/base/host_port_pair.h"
diff --git a/net/log/trace_net_log_observer.cc b/net/log/trace_net_log_observer.cc
index b5669a353ca70..385bb4cf314c6 100644
--- a/net/log/trace_net_log_observer.cc
+++ b/net/log/trace_net_log_observer.cc
@@ -12,7 +12,7 @@
 
 #include "base/check.h"
 #include "base/json/json_writer.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "base/values.h"
 #include "net/log/net_log_entry.h"
 #include "net/log/net_log_event_type.h"
diff --git a/net/log/trace_net_log_observer.h b/net/log/trace_net_log_observer.h
index eda1f92cc248c..c673174e02a17 100644
--- a/net/log/trace_net_log_observer.h
+++ b/net/log/trace_net_log_observer.h
@@ -7,7 +7,7 @@
 
 #include "base/memory/raw_ptr.h"
 #include "base/memory/weak_ptr.h"
-#include "base/trace_event/trace_log.h"
+#include "base/trace_event/base_tracing.h"
 #include "net/base/net_export.h"
 #include "net/log/net_log.h"
 
diff --git a/net/nqe/network_quality_estimator.cc b/net/nqe/network_quality_estimator.cc
index f01003fb857b9..9c0065407a1cb 100644
--- a/net/nqe/network_quality_estimator.cc
+++ b/net/nqe/network_quality_estimator.cc
@@ -28,7 +28,7 @@
 #include "base/task/single_thread_task_runner.h"
 #include "base/threading/thread_task_runner_handle.h"
 #include "base/time/default_tick_clock.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "build/build_config.h"
 #include "build/chromeos_buildflags.h"
 #include "net/base/features.h"
diff --git a/net/quic/quic_stream_factory.cc b/net/quic/quic_stream_factory.cc
index cfe01eedcf16a..6189281fa9edc 100644
--- a/net/quic/quic_stream_factory.cc
+++ b/net/quic/quic_stream_factory.cc
@@ -28,7 +28,7 @@
 #include "base/threading/thread_task_runner_handle.h"
 #include "base/time/default_clock.h"
 #include "base/time/default_tick_clock.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "base/values.h"
 #include "crypto/openssl_util.h"
 #include "net/base/address_list.h"
diff --git a/net/socket/client_socket_handle.cc b/net/socket/client_socket_handle.cc
index e0534344c6170..fece6c4333550 100644
--- a/net/socket/client_socket_handle.cc
+++ b/net/socket/client_socket_handle.cc
@@ -11,7 +11,7 @@
 #include "base/check_op.h"
 #include "base/compiler_specific.h"
 #include "base/notreached.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "net/base/net_errors.h"
 #include "net/base/trace_constants.h"
 #include "net/log/net_log_event_type.h"
diff --git a/net/socket/connect_job.cc b/net/socket/connect_job.cc
index 20bdfb1158e73..296a930d2a121 100644
--- a/net/socket/connect_job.cc
+++ b/net/socket/connect_job.cc
@@ -7,7 +7,7 @@
 #include <set>
 #include <utility>
 
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "net/base/connection_endpoint_metadata.h"
 #include "net/base/net_errors.h"
 #include "net/base/trace_constants.h"
diff --git a/net/socket/socket_posix.cc b/net/socket/socket_posix.cc
index 3f458f78a2a22..afacd89ceda08 100644
--- a/net/socket/socket_posix.cc
+++ b/net/socket/socket_posix.cc
@@ -17,7 +17,7 @@
 #include "base/logging.h"
 #include "base/posix/eintr_wrapper.h"
 #include "base/task/current_thread.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "build/build_config.h"
 #include "net/base/io_buffer.h"
 #include "net/base/ip_endpoint.h"
diff --git a/net/socket/socks5_client_socket.cc b/net/socket/socks5_client_socket.cc
index a3f842e6cdcb5..9ba9d9d6aa07e 100644
--- a/net/socket/socks5_client_socket.cc
+++ b/net/socket/socks5_client_socket.cc
@@ -12,7 +12,7 @@
 #include "base/format_macros.h"
 #include "base/strings/string_util.h"
 #include "base/sys_byteorder.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "net/base/io_buffer.h"
 #include "net/base/sys_addrinfo.h"
 #include "net/log/net_log.h"
diff --git a/net/socket/ssl_client_socket_impl.cc b/net/socket/ssl_client_socket_impl.cc
index a8e0aebcb1896..0aa23765cc8b1 100644
--- a/net/socket/ssl_client_socket_impl.cc
+++ b/net/socket/ssl_client_socket_impl.cc
@@ -29,7 +29,7 @@
 #include "base/strings/string_piece.h"
 #include "base/synchronization/lock.h"
 #include "base/threading/sequenced_task_runner_handle.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "base/values.h"
 #include "build/build_config.h"
 #include "crypto/ec_private_key.h"
diff --git a/net/socket/ssl_connect_job.cc b/net/socket/ssl_connect_job.cc
index 027f21c297d30..971b06a80c1ed 100644
--- a/net/socket/ssl_connect_job.cc
+++ b/net/socket/ssl_connect_job.cc
@@ -13,7 +13,7 @@
 #include "base/feature_list.h"
 #include "base/metrics/histogram_functions.h"
 #include "base/metrics/histogram_macros.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "net/base/connection_endpoint_metadata.h"
 #include "net/base/features.h"
 #include "net/base/host_port_pair.h"
diff --git a/net/socket/tcp_socket_posix.cc b/net/socket/tcp_socket_posix.cc
index 4e7a6edd9c05f..ff78d1575bc7f 100644
--- a/net/socket/tcp_socket_posix.cc
+++ b/net/socket/tcp_socket_posix.cc
@@ -22,7 +22,7 @@
 #include "base/strings/string_number_conversions.h"
 #include "base/strings/string_piece.h"
 #include "base/time/time.h"
-#include "base/trace_event/typed_macros.h"
+#include "base/trace_event/base_tracing.h"
 #include "build/build_config.h"
 #include "net/base/address_list.h"
 #include "net/base/io_buffer.h"
@@ -43,7 +43,6 @@
 #include "net/socket/socket_posix.h"
 #include "net/socket/socket_tag.h"
 #include "net/traffic_annotation/network_traffic_annotation.h"
-#include "third_party/perfetto/include/perfetto/tracing/string_helpers.h"
 
 #if BUILDFLAG(IS_ANDROID)
 #include "net/android/network_library.h"
diff --git a/net/socket/transport_connect_job.cc b/net/socket/transport_connect_job.cc
index 75397d9b8683d..ff36d225d13ba 100644
--- a/net/socket/transport_connect_job.cc
+++ b/net/socket/transport_connect_job.cc
@@ -16,7 +16,7 @@
 #include "base/notreached.h"
 #include "base/threading/thread_task_runner_handle.h"
 #include "base/time/time.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "net/base/features.h"
 #include "net/base/host_port_pair.h"
 #include "net/base/ip_endpoint.h"
diff --git a/net/socket/udp_socket_posix.cc b/net/socket/udp_socket_posix.cc
index d924fe6433fc8..d393fbad77119 100644
--- a/net/socket/udp_socket_posix.cc
+++ b/net/socket/udp_socket_posix.cc
@@ -34,7 +34,7 @@
 #include "base/rand_util.h"
 #include "base/task/current_thread.h"
 #include "base/task/thread_pool.h"
-#include "base/trace_event/typed_macros.h"
+#include "base/trace_event/base_tracing.h"
 #include "build/chromeos_buildflags.h"
 #include "net/base/features.h"
 #include "net/base/io_buffer.h"
@@ -54,7 +54,6 @@
 #include "net/socket/socket_tag.h"
 #include "net/socket/udp_net_log_parameters.h"
 #include "net/traffic_annotation/network_traffic_annotation.h"
-#include "third_party/perfetto/include/perfetto/tracing/string_helpers.h"
 
 #if BUILDFLAG(IS_ANDROID)
 #include "base/native_library.h"
diff --git a/net/spdy/spdy_session.cc b/net/spdy/spdy_session.cc
index 5d6354ac2a488..53b17d41c2d3f 100644
--- a/net/spdy/spdy_session.cc
+++ b/net/spdy/spdy_session.cc
@@ -29,8 +29,8 @@
 #include "base/task/single_thread_task_runner.h"
 #include "base/threading/thread_task_runner_handle.h"
 #include "base/time/time.h"
+#include "base/trace_event/base_tracing.h"
 #include "base/trace_event/memory_usage_estimator.h"
-#include "base/trace_event/trace_event.h"
 #include "base/values.h"
 #include "net/base/features.h"
 #include "net/base/proxy_server.h"
diff --git a/net/spdy/spdy_session_pool.cc b/net/spdy/spdy_session_pool.cc
index f44346a0cb937..ad0acd409eb37 100644
--- a/net/spdy/spdy_session_pool.cc
+++ b/net/spdy/spdy_session_pool.cc
@@ -14,7 +14,7 @@
 #include "base/ranges/algorithm.h"
 #include "base/strings/stringprintf.h"
 #include "base/threading/thread_task_runner_handle.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "base/values.h"
 #include "build/build_config.h"
 #include "net/base/ip_endpoint.h"
diff --git a/net/test/embedded_test_server/android/embedded_test_server_android.cc b/net/test/embedded_test_server/android/embedded_test_server_android.cc
index 08a9c45384fa5..1537f1c072138 100644
--- a/net/test/embedded_test_server/android/embedded_test_server_android.cc
+++ b/net/test/embedded_test_server/android/embedded_test_server_android.cc
@@ -10,7 +10,7 @@
 #include "base/bind.h"
 #include "base/files/file_path.h"
 #include "base/test/test_support_android.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "net/net_test_jni_headers/EmbeddedTestServerImpl_jni.h"
 
 using base::android::JavaParamRef;
diff --git a/net/test/embedded_test_server/controllable_http_response.cc b/net/test/embedded_test_server/controllable_http_response.cc
index 04980e10f6aea..f56bd65e2144f 100644
--- a/net/test/embedded_test_server/controllable_http_response.cc
+++ b/net/test/embedded_test_server/controllable_http_response.cc
@@ -8,7 +8,7 @@
 #include "base/check_op.h"
 #include "base/strings/stringprintf.h"
 #include "base/threading/thread_task_runner_handle.h"
-#include "base/trace_event/typed_macros.h"
+#include "base/trace_event/base_tracing.h"
 #include "net/test/embedded_test_server/http_response.h"
 
 namespace net::test_server {
diff --git a/net/url_request/url_request_http_job.cc b/net/url_request/url_request_http_job.cc
index 0a622f7f9948d..2378eba1c38b3 100644
--- a/net/url_request/url_request_http_job.cc
+++ b/net/url_request/url_request_http_job.cc
@@ -30,7 +30,7 @@
 #include "base/task/single_thread_task_runner.h"
 #include "base/threading/thread_task_runner_handle.h"
 #include "base/time/time.h"
-#include "base/trace_event/trace_event.h"
+#include "base/trace_event/base_tracing.h"
 #include "base/types/optional_util.h"
 #include "base/values.h"
 #include "build/build_config.h"
diff --git a/url/gurl.cc b/url/gurl.cc
index 7bec48c47abfc..23902e4640082 100644
--- a/url/gurl.cc
+++ b/url/gurl.cc
@@ -15,8 +15,8 @@
 #include "base/no_destructor.h"
 #include "base/strings/string_piece.h"
 #include "base/strings/string_util.h"
+#include "base/trace_event/base_tracing.h"
 #include "base/trace_event/memory_usage_estimator.h"
-#include "third_party/perfetto/include/perfetto/tracing/traced_value.h"
 #include "url/url_canon_stdstring.h"
 #include "url/url_util.h"
 
diff --git a/url/gurl.h b/url/gurl.h
index 13f443aeadcab..9c53c6562b28e 100644
--- a/url/gurl.h
+++ b/url/gurl.h
@@ -15,7 +15,7 @@
 #include "base/debug/alias.h"
 #include "base/debug/crash_logging.h"
 #include "base/strings/string_piece.h"
-#include "third_party/perfetto/include/perfetto/tracing/traced_value_forward.h"
+#include "base/trace_event/base_tracing_forward.h"
 #include "url/third_party/mozilla/url_parse.h"
 #include "url/url_canon.h"
 #include "url/url_canon_stdstring.h"
diff --git a/url/origin.cc b/url/origin.cc
index 48b5ca0cceb60..bcd10c8a46942 100644
--- a/url/origin.cc
+++ b/url/origin.cc
@@ -21,8 +21,8 @@
 #include "base/pickle.h"
 #include "base/strings/strcat.h"
 #include "base/strings/string_piece.h"
+#include "base/trace_event/base_tracing.h"
 #include "base/unguessable_token.h"
-#include "third_party/perfetto/include/perfetto/tracing/traced_value.h"
 #include "url/gurl.h"
 #include "url/scheme_host_port.h"
 #include "url/url_constants.h"
diff --git a/url/origin.h b/url/origin.h
index 02a7bf6e0f1a9..a388eb9985dc4 100644
--- a/url/origin.h
+++ b/url/origin.h
@@ -15,12 +15,12 @@
 #include "base/debug/crash_logging.h"
 #include "base/strings/string_piece_forward.h"
 #include "base/strings/string_util.h"
+#include "base/trace_event/base_tracing_forward.h"
 #include "base/unguessable_token.h"
 #include "build/build_config.h"
 #include "build/buildflag.h"
 #include "ipc/ipc_param_traits.h"
 #include "third_party/abseil-cpp/absl/types/optional.h"
-#include "third_party/perfetto/include/perfetto/tracing/traced_value_forward.h"
 #include "url/scheme_host_port.h"
 
 #if BUILDFLAG(IS_ANDROID)
-- 
2.39.0.314.g84b9a713c41-goog

